std-bits/table/src/main/java/org/enso/table/excel/xssfreader/XSSFReaderWorkbook.java

```diff
@@ -0,0 +1,299 @@
+    public String getNamespaceURI(String prefix) {
+      if (prefix == null) {
+        throw new IllegalArgumentException("prefix cannot be null");
+      }
+      return prefix.equals("ss") ? XSSFRelation.NS_SPREADSHEETML : XMLConstants.NULL_NS_URI;
+    }
+
+    @Override
+    public String getPrefix(String namespaceURI) {
+      if (namespaceURI == null) {
+        throw new IllegalArgumentException("namespaceURI cannot be null");
+      }
+      return namespaceURI.equals(XSSFRelation.NS_SPREADSHEETML) ? "ss" : null;
+    }
+
+    @Override
+    public Iterator<String> getPrefixes(String namespaceURI) {
+      if (namespaceURI == null) {
+        throw new IllegalArgumentException("namespaceURI cannot be null");
+      }
+      return namespaceURI.equals(XSSFRelation.NS_SPREADSHEETML)
+          ? Collections.singleton("ss").iterator()
+          : Arrays.stream(new String[0]).iterator();
+    }
+  }
+
+  public static final String WORKBOOK_CONFIG_XPATH = "/ss:workbook/ss:workbookPr";
+  public static final String SHEET_NAME_XPATH = "/ss:workbook/ss:sheets/ss:sheet";
+  public static final String NAMED_RANGE_XPATH = "/ss:workbook/ss:definedNames/ss:definedName";
+
+  private final String path;
+
+  private boolean use1904DateSystemFlag = false;
+  private List<SheetInfo> sheetInfos;
+  private Map<String, SheetInfo> sheetInfoMap;
+  private Map<String, NamedRange> namedRangeMap;
+
+  private boolean readShared = false;
+  private SharedStrings sharedStrings;
+  private XSSFReaderFormats styles;
+
+  public XSSFReaderWorkbook(String path) throws IOException {
+    this.path = path;
+
+    // Read the workbook data
+    this.readWorkbookData();
+  }
+
+  public String getPath() {
+    return path;
+  }
+
+  void withReader(Consumer<XSSFReader> action) throws IOException {
+    try (var pkg = OPCPackage.open(path, PackageAccess.READ)) {
+      var reader = new XSSFReader(pkg);
+      action.accept(reader);
+    } catch (OpenXML4JException e) {
+      throw new IOException(
+          "Invalid format encountered when opening the file " + path + " as XLSX.",
+          e);
+    }
+  }
+
+  private record SheetInfo(int index, int sheetId, String name, String relID, boolean visible) {}
+
+  private record NamedRange(String name, String formula) {}
+
+  private void readWorkbookData() throws IOException {
+    withReader(
+        rdr -> {
+          try {
+            var workbookData = rdr.getWorkbookData();
+            var workbookDoc = DocumentHelper.readDocument(workbookData);
+
+            // Read the Workbook settings
+            var workbookXPath = compileXPathWithNamespace(WORKBOOK_CONFIG_XPATH);
+            var workbookNode = (Node) workbookXPath.evaluate(workbookDoc, XPathConstants.NODE);
+            if (workbookNode != null) {
+              var date1904 = workbookNode.getAttributes().getNamedItem("date1904");
+              use1904DateSystemFlag = date1904 != null && "1".equals(date1904.getNodeValue());
+            }
```

Comment:
nitpick, but for better code readability/structure I'd suggest to split off every of these 'named blocks' (read settings, read sheets, read Named Ranges) into helper functions. It feels like it would make it easier to track each one if they are built from slightly smaller building blocks VS a single huge function